### PR TITLE
Addressing concerns in #6 

### LINKED ISSUE #6: Consolidate generation to a single plugin version and up-to-date libraries
### Description
Currently, code generation is distinct for each of the modules in this repository. Different versions of libraries/plugins are being used for different modules and they are not the latest/best versions.

1. The modules should all be updated to use the same set of libraries and tools for jaxb generation
2. Plugins and libraries should be declared and managed in the reactor pom
3. Each module should include little or no plugin configuration, except in cases where specific configuration parameters need to be passed to the plugins

### Steps to Reproduce
N/A as this is an update to the build system.

#### Expected behavior:
The resulting artifacts should be substitutable into DDF.

#### Actual behavior:
N/A

#### Reproduces how often:
N/A

### Version
1.0.0

### Additional Information
Any additional information, configuration or data that might be necessary
to reproduce the issue.

### PR DESCRIPTION
**Pull request by Kjames5269**
### Requirements
* Filling out the template is required. Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

@stustison @coyotesqrl 
@bakejeyner @blen-desta 

### Description of the Change

Cleaning up all the pom.xml files.

### Alternate Designs

Two main decisions were made while refactoring. First was to get all the generic configurations used by every child pom into the root pom. The second was to get as much use out of default configurations as possible provided by the `jaxb2-maven-plugin` as well as removing unnecessary tags along the way. An example of this is `<forceRegenerate>`. It has been deprecated as of version `0.12.1` of the plugin and claimed to be unneeded. However, there's a known issue where if multiple `<execution>` blocks are run with the same target directory then it will fail to generate any code past the first `<execution>` block if `<forceRegeneration>` is not present. The work around for the future to avoid using a deprecated tag is to use separate target directories which has been added in the latest commit. 

### Benefits

Maintainability and consistency. 

### Possible Drawbacks

None-

### Verification Process

Shamelessly copied from [here](https://github.com/codice/ddf/pull/3898)
1. Full [DDF](https://github.com/codice/ddf) CI build must pass
2. Install DDF and test affected functionality
   * Create CSW sources and ensure they work correctly
   * Create WFS 1.0, 1.1, and 2.0 sources and ensure they work correctly
   * Ensure security - CAS and XACML in particular - still work
   * Ensure that metacard transformations (ingests, etc.) still work correctly

### Applicable Issues

Fixes: #6

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -139,6 +150,29 @@
...
+                <plugin>
+                    <groupId>org.jvnet.jaxb2.maven2</groupId>
+                    <artifactId>maven-jaxb2-plugin</artifactId>
+                    <version>${maven-jaxb2-plugin.version}</version>
+                    <configuration>
+                        <plugins combine.children="append">
+                            <plugin>
+                                <groupId>org.jvnet.jaxb2_commons</groupId>
+                                <artifactId>jaxb2-basics</artifactId>
+                                <version>${jaxb2.version}</version>
```
Comment (coyotesqrl) [inline pom.xml]: ✏️ Maybe we could make this a little more distinct from `jaxb.version` by naming it `jvnet.jaxb2.version`?

**Review by coyotesqrl — APPROVED** [review]
This has all been verified against DDF, I assume?

File: pom.xml (hunk 1 above)
Comment (Kjames5269) [reply pom.xml]: Good idea, I'm also gonna check to see if there's consistency issues in naming between here and DDF with anything I added in.

**Comment by Kjames5269** [comment]
Yeah, I ran a full build on DDF and it passed. On top of that, I was able to ingest and query over CSW soruces. However, I did have trouble with the geoserver, but haven't verified if it's a preexisting problem on master or something these changes introduced.

**Review by blen-desta — APPROVED** [review]

**Review by bakejeyner — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -30,22 +30,33 @@
...
         <ows-v_1_0_0-schema.version>1.1.0</ows-v_1_0_0-schema.version>
-        <jaxb2.version>1.11.1</jaxb2.version>
         <maven-source-plugin.version>3.0.1</maven-source-plugin.version>
 
+        <!-- Neither of the below modules use semantic versioning correctly -->
+        <!-- Even though it's only a minor version bump <jaxb>2.3.0</jaxb> is a breaking change -->
+        <!-- and cannot be updated until our schemas use a newer version of JAXB2 -->
+        <!-- <jaxb2>0.11.0</jaxb2> also uses 2.3.0 -->
+        <jaxb.version>2.2.11</jaxb.version>
+        <jaxb2.version>0.10.0</jaxb2.version>
```
Comment (stustison) [inline pom.xml]: ❓ why did this change from 0.11.1 to 0.10.0 (yeah I know the 1.11.x stuff is wrong).

Comment (stustison) [reply pom.xml]: Seems like we should use 0.11.1 or even 0.12.0 if possible.

Comment (Kjames5269) [reply pom.xml]: I don't actually know where 0.11.1 came from to be honest. It doesn't seem like it was being used at all in this repo. ddf-master is still using 0.9.4. @coyotesqrl might know.

Comment (coyotesqrl) [reply pom.xml]: I do not recall where that version came from. I would definitely prefer the latest/greatest if possible.

Comment (Kjames5269) [reply pom.xml]: I just verified that we can't update with our current schemas. The newer versions of the `jaxb2-basics` plugin attach what I'll call method 2's to everything. (`toString2()`). Our schemas call methods that are no longer created with the correct parameters giving us a type mismatch
```
[ERROR]     method ToString.appendFields(ObjectLocator,StringBuilder,ToStringStrategy) is not applicable
[ERROR]       (argument mismatch; ToStringStrategy2 cannot be converted to ToStringStrategy)
```

**Review by stustison — APPROVED** [review]

**Comment by bakejeyner** [comment]
Hero ✔️

**Pull request merged**